### ISSUE TITLE
CRD scrapeConfigs doesn't honour scheme: https and tlsConfig

### ISSUE THREAD

**Issue by vladimir9876** [prometheus-operator/prometheus-operator#5754]
**What happened?**
I've deployed:

<pre>
  scheme: HTTPS
  tlsConfig:
        insecureSkipVerify: true
</pre>

within my ScrapeConfig CRD but it's not reflected in the prometheus config.
scheme stays http, and tlsConfig is not present.

**Did you expect to see something different?**

Yes.
I expected these directives to be applied to the prometheus config.

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* Prometheus Operator version:

v0.66.0
    
    

* Kubernetes version information:

v1.26.4+k3s1
    

* Kubernetes cluster kind:

k3s (part of rancher desktop)

* Manifests:

```
apiVersion: monitoring.coreos.com/v1alpha1
kind: ScrapeConfig
metadata:
  name: ldap
  namespace: monitoring
  labels:
    test: value
spec:
  scheme: HTTPS
  staticConfigs:
    - targets:
      - myserver.com:9330
  tlsConfig:
    insecureSkipVerify: true
```

* Prometheus Operator Logs:

```
insert Prometheus Operator logs relevant to the issue here
```

**Anything else we need to know?**:

**Comment by slashpai** [comment]
This feature is not available in `0.66` :). This will be out in `0.67`. Feature is merged in master branch only.
If you want to try you need to use crd's from master branch and build image from master branch (`make image`) to try.

**Comment by vladimir9876** [comment]
Thank you Jayapriya.

I was confused as 0.66 CRD states these options and i was able to apply 
the manifest.

Regards,

Vladimir

On 2023-07-20 14:00, Jayapriya Pai wrote:

> This feature is not available in 0.66 :). This will be out in 0.67. 
> Feature is merged in master branch only.
> If you want to try you need to use crd's from master branch and build 
> image from master branch (make image) to try.
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://github.com/prometheus-operator/prometheus-operator/issues/5754#issuecomment-1643884429
[2]
